snuba/query/expressions.py

```diff
@@ -110,10 +121,67 @@ def visit_lambda(self, exp: Lambda) -> TVisited:
         raise NotImplementedError
 
 
+# DictifiedExpr = Dict[str, Union[None, str, int, Dict[str, Any]]]
+
+# NOTE (Vlad): recursive type definitions are not yet supported in mypy
+# https://github.com/python/mypy/issues/731 hence sub dictionaries
+# are expressed as Dict[str, Any]
+DictifiedExpr = Dict[
+    str, Union[None, str, Dict[str, Any], List[Union[str, Dict[str, Any]]]]
+]
+
+
+class DictifyVisitor(ExpressionVisitor[DictifiedExpr]):
+    """Visitor implementation to turn an expression into a dictionary format"""
+
+    def visit_literal(self, exp: Literal) -> DictifiedExpr:
+        d = asdict(exp)
+        d["__name__"] = exp.__class__.__name__
+        return d
+
+    def visit_column(self, exp: Column) -> DictifiedExpr:
+        d = asdict(exp)
+        d["__name__"] = exp.__class__.__name__
+        return d
+
+    def visit_subscriptable_reference(
+        self, exp: SubscriptableReference
+    ) -> DictifiedExpr:
+        return {
+            "__name__": exp.__class__.__name__,
+            "column": exp.column.accept(self),
+            "key": exp.key.accept(self),
+        }
+
+    def visit_function_call(self, exp: FunctionCall) -> DictifiedExpr:
+        return {
+            "__name__": exp.__class__.__name__,
+            "function_name": exp.function_name,
+            "parameters": [param.accept(self) for param in exp.parameters],
+        }
+
+    def visit_curried_function_call(self, exp: CurriedFunctionCall) -> DictifiedExpr:
+        return {
+            "__name__": exp.__class__.__name__,
+            "internal_function": exp.internal_function.accept(self),
+            "parameters": [param.accept(self) for param in exp.parameters],
+        }
+
+    def visit_argument(self, exp: Argument) -> DictifiedExpr:
+        return {"__name__": exp.__class__.__name__, "name": exp.name}
+
+    def visit_lambda(self, exp: Lambda) -> DictifiedExpr:
+        return {
+            "__name__": exp.__class__.__name__,
+            "transformation": exp.transformation.accept(self),
+            "parameters": [param for param in exp.parameters],
+        }
+
+
 OptionalScalarType = Union[None, bool, str, float, int, date, datetime]
 
 
-@dataclass(frozen=True)
+@dataclass(frozen=True, repr=False)
```

Comment:
Why do you not have a `repr` on these anymore?

Reply:
dataclasses have a default repr, you need to explicitly say you're using your own so that it uses your implementation